vanilla-tests/src/main/java/fixtures/streamstyleserialization/models/Salmon.java

```diff
@@ -132,13 +118,14 @@ public void validate() {
     @Override
     public JsonWriter toJson(JsonWriter jsonWriter) throws IOException {
         jsonWriter.writeStartObject();
-        jsonWriter.writeFloatField("length", getLength());
-        jsonWriter.writeStringField("species", getSpecies());
-        jsonWriter.writeArrayField("siblings", getSiblings(), (writer, element) -> writer.writeJson(element));
-        jsonWriter.writeStringField("fishtype", this.fishtype);
+        toJsonShared(jsonWriter);
+        return jsonWriter.writeEndObject();
+    }
+
+    void toJsonShared(JsonWriter jsonWriter) throws IOException {
+        super.toJsonShared(jsonWriter);
```

Comment:
Is this only for the case that super class and sub class in the same package? Can super class and sub class be in different packages? Then `super.toJsonShared` could not be called.